src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -260,7 +260,7 @@ public override void VisitBinaryExpression(BinaryExpressionSyntax node)
         public override void VisitDeclarationExpression(DeclarationExpressionSyntax node)
         {
             var argumentSyntax = (ArgumentSyntax)node?.Parent;
-            var argumentListSyntax = (BaseArgumentListSyntax)argumentSyntax?.Parent;
+            var argumentListSyntax = argumentSyntax.Parent as BaseArgumentListSyntax;
```

Comment:
There are five at the moment. See `OutVarTests.OutVarDeconstruction_02`, `OutVarTests.OutVarDeconstruction_03`, `OutVarTests.OutVarDeconstruction_04`, `PatternSwitchTests.TupleInPattern`, and `LocalFunctionTests.NoOperator` is also affected.